9-type-aliases-redux/9-type-aliases-redux.md

```diff
@@ -52,15 +65,57 @@ as a type variable hole. Like other holes, each type variable hole has a
 metavariable.
 
 ## Kinds
+
 We will need kinds to distinguish types from type-level holes, such as `TyVarHole`.
 We define a new type, `Kind.t`, as follows:
 ```
 type Kind.t = KHole
             | Type
+            | Singleton(HTyp.t)
 ```
 
+## Unexpanded Expressions
+
+To support type aliases, we need to add a new line item to the language and the proposed syntax is
+
+`TyAliasLine(TPat.t, UHTyp.t)` for abstract syntax
+
+`type t = <type> in e`  for GUI syntax
+
+### `ZExp.re` :
+
+By considering the cursor positions of `TyAlias`, we need to add two more zline items which are
+
+`TyAliasLineT(TPat.t, ZTyp.t)` (when cursor is on the type)
+
+`TyAliasLineP(ZTPat.t, UHTyp.t)` (when cursor is on the type pattern)
+
+
+## Expanded Expressions
+
+So that we can still assign a type properly to the resulting `DExp.t` we add the following new syntax:
+
+`TyBindings(TyVarCtx.t, DHExp.t)` for abstract syntax
```

Comment:
not sure why this needs to be a tyvarctx -- I think we can just turn each type alias line into a corresponding type alias expression in dhexp

Reply:
I was thinking we could avoid having to deal with patterns in DHExp if we did it this way and this will scale if/when type-patterns get more complicated.

I can't think of a way to, in a single DHExp, capture all the possibilities of holes in different places of the UHExp type alias without copying all the rules over (or is this what you're suggesting?)